include/multiphase/private/fd_operators_inc.h

```diff
@@ -307,43 +418,42 @@ accumulateMomentumForcesOnPatchConstantCoefficient(SAMRAI::tbox::Pointer<SAMRAI:
             (convertToThs((*thn_data)(idx_c_up)) * ((*us_data)(upper_x_idx) - (*us_data)(lower_x_idx)) -
              convertToThs((*thn_data)(idx_c_low)) * ((*us_data)(u_x_idx) - (*us_data)(l_x_idx)));
         double drag_s = -xi / nu_s * thn_lower * convertToThs(thn_lower) * ((*us_data)(idx) - (*un_data)(idx));
-        double pressure_s = -convertToThs(thn_lower) / dx[1] * ((*p_data)(idx_c_up) - (*p_data)(idx_c_low));
-        (*A_us_data)(idx) = D_u * (ddy_Ths_dy_us + ddx_Ths_dx_us + ddx_Ths_dy_vs + ddy_Ths_dx_vs + drag_s) +
-                            D_p * (pressure_s) + C * convertToThs(thn_lower) * (*us_data)(idx);
+        (*F_us_data)(idx) = D_u * (ddy_Ths_dy_us + ddx_Ths_dx_us + ddx_Ths_dy_vs + ddy_Ths_dx_vs + drag_s) +
+                            C * convertToThs(thn_lower) * (*us_data)(idx);
     }
 }
 
+// Accumulate the momentum forces with thn only provided at cell centers.
 inline void
-accumulateMomentumForcesOnPatchVariableDrag(SAMRAI::tbox::Pointer<SAMRAI::hier::Patch<NDIM>> patch,
-                                            const int A_un_idx,
-                                            const int A_us_idx,
-                                            const int p_idx,
-                                            const int un_idx,
-                                            const int us_idx,
-                                            const int thn_idx,
-                                            const MultiphaseParameters& params,
-                                            const double C,
-                                            const double D_u,
-                                            const double D_p)
+accumulateMomentumWithoutPressureOnPatchConstantCoefficient(SAMRAI::tbox::Pointer<SAMRAI::hier::Patch<NDIM>> patch,
+                                                            const int F_un_idx,
+                                                            const int F_us_idx,
+                                                            const int un_idx,
+                                                            const int us_idx,
+                                                            const int thn_idx,
+                                                            const MultiphaseParameters& params,
+                                                            const double C,
+                                                            const double D_u)
 {
 #ifndef NDEBUG
-    TBOX_ASSERT(params.isVariableDrag());
+    TBOX_ASSERT(!params.isVariableDrag());
 #endif
```

Comment:
We should start providing more useful error messages than just throwing assertions.